### ISSUE TITLE
On the rail surface defect detection dataset, the detection accuracy of YOLO26n is much worse than that of YOLO11n.

### ISSUE THREAD

**Issue by wang-yu-tao** [ultralytics/ultralytics#24225]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

On the rail surface defect detection dataset, the detection accuracy of YOLO26n is much worse than that of YOLO11n.

<img width="780" height="219" alt="Image" src="https://github.com/user-attachments/assets/31085928-2941-4afd-8621-10554ec05c6a" />

<img width="850" height="218" alt="Image" src="https://github.com/user-attachments/assets/da12d1fb-98c0-4f00-849d-46cb979b852f" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wang-yu-tao, thank you for your interest in Ultralytics 🚀! This is an automated response to help gather the details needed for review, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a custom training/performance ❓ question, please provide as much information as possible to help with comparison, including dataset image examples, training logs, model arguments, dataset YAML, augmentation settings, input image size, number of epochs, hardware details, and whether both YOLO11n and YOLO26n were trained under the exact same conditions. If possible, please also share a small reproducible training example so the behavior can be reviewed more effectively 🔍.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You should try training with latest Ultralytics

**Comment by wang-yu-tao** [comment]
> You should try training with latest Ultralytics你应该试试用最新的 Ultralytics 训练

When 8.4.6 and 8.4.27 were the latest Ultralytics versions, the YOLO26n I ran performed worse than YOLO11n. I haven't trained on 8.4.37 yet because I saw that the release logs from 8.4.27 to 8.4.37 did not make any changes to the network architecture or optimize the detection network.

**Comment by glenn-jocher** [comment]
Even without architecture changes, updates to training defaults, losses, augmentations, and bug fixes can still change final mAP, so please rerun a strict apples-to-apples comparison on `ultralytics==8.4.37` with the same split, seed, `imgsz`, epochs, and hyp settings for both `yolo11n.pt` and `yolo26n.pt`, then share the exact train commands, `yolo checks`, and full `results.csv` using the [MRE guide](https://docs.ultralytics.com/help/minimum-reproducible-example/); for very small rail defects, also test a larger `imgsz` or `yolo26s.pt`, since `n` models can underfit tiny targets.

**Comment by Y-T-G** [comment]
How many training images do you have?

**Comment by wang-yu-tao** [comment]
> How many training images do you have?
This is my training data information.
There are a total of 2,000 training images.

<img width="1125" height="1125" alt="Image" src="https://github.com/user-attachments/assets/306098a1-f69d-4284-ba13-982795ed646e" />

**Comment by wang-yu-tao** [comment]
> How many training images do you have?

All experiments were conducted on ultralytics=8.4.27, without using pretrained weights, and trained for 500 epochs.

1. When using YOLO26n:
   1. With lr0=0.01, optimizer=MuSGD, and end2end=True, mAP50=0.778, And when using MuSGD, the training time is longer.
   3. With lr0=0.001, optimizer=AdamW, and end2end=True, mAP50=0.764
   4. With lr0=0.001, optimizer=AdamW, and end2end=False, mAP50=0.809
2. When using YOLO11n:
   With lr0=0.001 and optimizer=AdamW, mAP50=0.801

**Comment by glenn-jocher** [comment]
Your results suggest this is mainly a training-setup effect, not a clear model regression: on your dataset `yolo26n` is already slightly above `yolo11n` when you keep `optimizer=AdamW` and use `end2end=False` (`0.809` vs `0.801`), so for this small 2k-image tiny-defect task I’d treat `end2end=True` as the likely cause of the drop and compare both again with identical `optimizer`, `lr0`, `imgsz`, and preferably pretrained weights or `yolo26s.pt`; the [YOLO26 vs YOLO11 comparison](https://docs.ultralytics.com/compare/yolo26-vs-yolo11/) also notes YOLO26’s small-object gains depend on the training setup.
